subprojects/docs/src/docs/userguide/migrating_from_ant.adoc

```diff
@@ -0,0 +1,328 @@
++
+The downside is that you lose the advantages of Gradle's conventions, many of its plugins, and many of its important features like incremental build, proper multi-project support, the build cache, and more.
+ 2. Creating the Gradle build from scratch.
++
+This sounds intimidating, but you may be surprised at how much heavy lifting Gradle's plugins can do. You will often find that large swathes of an Ant build are unnecessary in the equivalent Gradle one. The big advantage of this approach is that you gain access to all of Gradle's most important features.
++
+The downside is that many builds will have behavior that can't be replicated by existing plugins, so the migration will take time and effort. You also need a good understanding of what the Ant build is doing in order to match the behavior in Gradle accurately.
+
+With those approaches in mind, the first thing you should decide is whether your project is a good fit for any of the standard Gradle project types. We expect that the vast majority of Ant builds are for JVM-based projects that do the same sorts of things:
+
+ * Compile classes
+ * Run tests
+ * Create JAR packages
+ * Produce Javadoc API documentation
+ * Manage dependencies
+
+There are other common tasks that depend on whether the build produces a library, an application, a webapp, and so on, for which Gradle has specific plugins.
+
+If you have such a build, we recommend that you take the approach of creating a Gradle build from scratch using the appropriate plugins — such as the Java Plugin as described in the chapter <<building_java_projects#building_java_projects,Building Java & JVM projects>> — and configuring them as necessary. You might even want to consider changing the directory structure of your project to match Gradle's conventions, e.g. `src/main/java` for Java source files, although you can usually retain the structure and inform Gradle about it via <<building_java_projects#sec:custom_java_source_set_paths,source-set configuration>>.
+
+If your build isn't a good fit for the standard plugins, then you're probably better off using `ant.importBuild()` and updating the build piecemeal to take advantage of features like incremental build. link:{javadocPath}org/gradle/api/AntBuilder.html#importBuild-java.lang.Object-org.gradle.api.Transformer-[Variant that allows you to change the name of tasks on import].
+
+Whichever approach you take, there are some common scenarios you will likely have to deal with during the migration, and we cover those in the rest of the chapter.
+
+== Managing dependencies
+
+Ant builds typically take one of two approaches to dealing with dependencies:
+
+ * Storing them with the project in a local "lib" directory
+ * Using https://ant.apache.org/ivy/[Ivy] to manage them
+
+Gradle can work with either situation with the appropriate configuration.
```

Comment:
Link to how supporting materials?

Reply:
Not sure what you mean, but I realise that this paragraph should properly lead into the following subsections, explaining that we will detail how to handle both scenarios.